client/client.go

```diff
@@ -0,0 +1,135 @@
+package client
+
+import (
+	"net/http"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"bytes"
+	"strings"
+	"fmt"
+	"time"
+	"io/ioutil"
+	"strconv"
+	"github.com/microdevs/missy/log"
+)
+// todo: implement token auth
+// MiSSy Client object. Target is the service that will be called
+type Client struct {
+	Target string
+	host string
+	port string
+}
+
+// Create a new Client for a target service
+func New(target string) *Client {
+	//todo: implement service discovery and set host and port
+	return &Client{Target: target}
+}
+
+// Generall Call function can handle all http methods, query string and json marshalling for posts
+func (c *Client) Call(method string, path string, query map[string]string, v interface{}) (*http.Response, error) {
+	//  create a reader in case we need to post or put
+	var dataReader *bytes.Reader
+
+	if method == "POST" || method == "PUT" {
+		dataByte, jsonErr := json.Marshal(v)
+		if jsonErr != nil {
+			return nil, errors.New("Failed to Marshal data to json")
+		}
+		dataReader = bytes.NewReader(dataByte)
+	}
+
+	// prepare the query string
+	queryString := ""
+	if len(query) > 0 {
+		queryString += "?"
+		for k,v := range query {
+			queryString += k+"="+v+"&"
+		}
+		queryString = strings.TrimRight(queryString, "&")
+	}
+
+	// build the url
+	url := "http://"+c.Target+":8080"+"/"+strings.TrimLeft(path, "/")+queryString
+	log.Debugf("Calling service %s with url %s", c.Target, url)
+	// create a net/http client
+	// todo: make timeout configurable
+	client := http.Client{Timeout: 10 * time.Second}
+	// make the differnt calls depending on the method
+	switch method {
+		case http.MethodGet:
+			return client.Get(url)
+		case http.MethodPost:
+			return client.Post(url, "application/json", dataReader)
```

Comment:
"application/json" this is something that bug me. I tried to find solution in standard package but didn't found any constant for this. It maybe not that much of a deal but maybe it will be good to start introduce common constants for reuse and typo error avoidenence?

Reply:
definitely a constant makes sense, I wanted to search for a proper constant myself but as it turns out there was none for content types, if we agree that the only exchange format for our services is JSON that there is not a lot of change, yet a constant makes total sense. probably later we can even allow to override the content type